k2/csrc/array_ops_test.cu

```diff
@@ -102,11 +108,11 @@ TEST(OpsTest, TransposeGpuTest) {
     // speed test for different data type
     // TODO(haowen): we may need to allocate different size of shared memory for
     // different data type to get the best performance
-    GpuTransposeTest<char>(1000, 2000, 100, true);
-    GpuTransposeTest<short>(1000, 2000, 100, true);
-    GpuTransposeTest<int>(1000, 2000, 100, true);
-    GpuTransposeTest<float>(1000, 2000, 100, true);
-    GpuTransposeTest<double>(1000, 2000, 100, true);
+//    GpuTransposeTest<char>(1000, 2000, 100, true);
```

Comment:
I'd prefer to leave these un-commented.  If it fails to compile, it fails.  I dont want an 'artificial' success.

Reply:
and why comment? all tests got passed with my last pr

Reply:
For this compile, the Transpose() gives an error that I have no clue. Need @qindazhu help here. I have to comment it to test other compile and test. The same condition also to one position, may you found the commented lines (totally < 6 lines).

Reply:
so can you please show me the build error for `Transpose()`? I'm definitely sure it gets compiled and run successfully with my last PR, so I guess there's some other code (between your PR and mine) causes the error